Basic cloning support

jaor opened the pull request:
This is a patch i had around that adds very basic cloning support.

ashenfad commented:
Looks good, but I'm seeing a failure with the dataset cloning test.  It seems the `:new_fields` values don't match.

jaor commented:
Sorry for the delay.  I've updated the PR with fixes for the failing unit tests, and i've also added a tiny bit of documentation and an upgrade for dependencies.

ashenfad commented:
Sorry - fell off my radar for a while.  Looks good, though.  Merged!
